src/Twig/RecordExtension.php

```diff
@@ -87,29 +92,34 @@ public function pager(Environment $env, Pagerfanta $records, string $template =
         return $env->render($template, $context);
     }
 
-    public function menu(Environment $env, string $template = '')
+    public function menu(Environment $env, ?string $name = null, string $template = '_sub_menu.twig', string $class = '', bool $withsubmenus = true): string
     {
-        // @todo See Github issue https://github.com/bolt/four/issues/253
-        return '[menu placeholder]';
+        $context = [
+            'menu' => $this->frontendMenuBuilder->getMenu($name),
+            'class' => $class,
+            'bool' => $withsubmenus,
+        ];
+
+        return $env->render($template, $context);
     }
 
     public static function excerpt(string $text, int $length = 100): string
     {
         return Excerpt::getExcerpt($text, $length);
     }
 
-    public function sidebarmenu($pretty = false)
+    public function sidebarmenu($pretty = false): string
     {
-        if (! $this->menu) {
+        if (! $this->sidebarMenu) {
             $menuArray = $this->menuBuilder->getMenu();
             $options = $pretty ? JSON_PRETTY_PRINT : 0;
-            $this->menu = json_encode($menuArray, $options);
+            $this->sidebarMenu = json_encode($menuArray, $options);
```

Comment:
why menu is saved to a property? does it need runtime cache?

Reply:
Yes. It's quite heavy, because it literally goes through all contenttypes and fetches the latest 5 records of each. You can probably go back in the history and find where we've previously talked about this. :-)

![Screenshot 2019-04-01 at 06 50 31](https://user-images.githubusercontent.com/1833361/55304176-82c87880-544a-11e9-81f1-a4372aba08aa.png)

Reply:
Also, it's used at least twice on each call.